ui/packages/catalog-ui-search/src/main/webapp/dev/component/base-guide/base-guide.view.js

```diff
@@ -0,0 +1,108 @@
+const Marionette = require('marionette');
+import React from 'react';
+import {
+    render
+} from 'react-dom';
+import AceEditor from 'react-ace';
+import 'brace/mode/html';
+import 'brace/mode/less';
+import 'brace/mode/javascript';
+import 'brace/theme/tomorrow_night';
+const beautify = require('js-beautify');
+
+const renderAce = ({
+    where,
+    mode,
+    value,
+    view
+}) => {
+    where.innerHTML = '<div class="is-large-font"><span class="fa fa-refresh fa-spin"></span></div>';
+    view.timeout+=1000;
+    setTimeout(() => {
+        if (view.isDestroyed) {
+            return;
+        }
+        render( <
```

Comment:
✏️ The formatting in this section is a bit funky. Can we clean it up a bit? (normalize the `{\n value \n}` vs `{ value }` vs `{value}`, etc)

Reply:
Maybe like
```
render(<AceEditor
        mode = { mode }
        theme = "tomorrow_night"
        value = { value }
        ...
        width = '100%'/>, where)
```

Reply:
I wonder if we should consider using [`reactToMarionette`](https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/transmute/index.js#L36) here.